tools/gcs-bucket-mover/gcs_bucket_mover/configuration.py

```diff
@@ -0,0 +1,151 @@
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+#
+# Any software provided by Google hereunder is distributed "AS IS", WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, and is not intended for production use.
+"""Class to hold all of the config values set up on initial script run."""
+
+import os
+
+from google.cloud import storage
+from google.oauth2 import service_account
+
+
+class Configuration(object):
+    """Class to hold all of the config values set up on initial script run."""
+
+    # pylint: disable=too-many-instance-attributes
+    # I don't think this class should be split into multiple config classes because all of
+    # the attributes are closely related.
+
+    def __init__(self, conf):
+        """Load the values from config.sh into global variables.
+
+        Set up the credentials and storage clients.
+
+        Args:
+            conf: the argparser parsing of command line options
+        """
+
+        self._source_service_account_json = os.getenv(
+            'GCP_SOURCE_PROJECT_SERVICE_ACCOUNT_KEY')
+        self._target_service_account_json = os.getenv(
+            'GCP_TARGET_PROJECT_SERVICE_ACCOUNT_KEY')
+
+        if (self._source_service_account_json is
+                None) or (self._target_service_account_json is None):
+            raise SystemExit(
+                'Missing some environment variables. Do you need to edit and source the config.sh'
+                ' script?')
+
+        self._source_project_credentials = service_account.Credentials.from_service_account_file(
+            self._source_service_account_json)
+        self._target_project_credentials = service_account.Credentials.from_service_account_file(
+            self._target_service_account_json)
+        print 'Using the following service accounts for GCS credentials: '
+        print('Source Project - {}'.format(
+            self._source_project_credentials.service_account_email))
+        print('Target Project - {}\n'.format(
+            self._target_project_credentials.service_account_email))
+
+        self._source_storage_client = storage.Client.from_service_account_json(
+            self._source_service_account_json)
+        self._target_storage_client = storage.Client.from_service_account_json(
+            self._target_service_account_json)
+        self._target_service_account_email = self.__get_service_account_email(
+            self._target_storage_client)
+
+        # Load the values passed in on the command line
+        self._bucket_name = conf.bucket_name
+        self._temp_bucket_name = self.__get_temp_bucket_name(
+            self._bucket_name, conf.tempBucketName)
+        self._source_project = conf.source_project
+        self._target_project = conf.target_project
+
+    @property
```

Comment:
You should use public instance values instead of getters with the @property decorator unless they're actually needed.

Reply:
After I removed all of the properties, the class was nothing but __init__ setting a but of variables and so pylint complained it should be a namedtuple or something instead of a class. I refactored it into a data class using the attr module, although pylint still doesn't like it, but I think it's much better than changing it to a namedtuple.